### PR TITLE
DAOS-5434: weekly-test: ObjOpenBadParam: test_invalid_handle and test_closed_container_handle: Test died without reporting the status

### PR DESCRIPTION
**Pull request by cibervicho**
Tests 'test_invalid_container_handle' and 'test_closed_container_handle' were
failing not reporting the expected status because in the obj_layout_create()
function in object/cli_obj.c L308, when trying to get the pool from an
incorrect container handler, the test was aborting as soon as it noticed
that pool==NULL.
In order to fix this the D_ASSERT call was replaced with a D_GOTO call
instead, passing the expected '-DER_NO_HDL' return code. With this, the
'open()' function of the DaosObj class can raise the DaosApiError to the
test itself.

Signed-off-by: David Maldonado Moreno <david.maldonado.moreno@intel.com>

### CODE REVIEW

**Comment by cibervicho** [comment]
The test obj_open_bad_param passed successfully now:

2021-02-09 14:34:05,006 test             L0139 INFO | *** SETUP running on <LinuxDistro: name=centos, version=7, release=9, arch=x86_64> ***
2021-02-09 14:34:37,954 test             L0779 INFO | PASS 1-./object/obj_open_bad_param.py:ObjOpenBadParam.test_bad_obj_handle;hosts-pool-server_config-bc1b
2021-02-09 14:34:50,730 test             L0779 INFO | PASS 2-./object/obj_open_bad_param.py:ObjOpenBadParam.test_invalid_container_handle;hosts-pool-server_config-bc1b
2021-02-09 14:35:03,334 test             L0779 INFO | PASS 3-./object/obj_open_bad_param.py:ObjOpenBadParam.test_closed_container_handle;hosts-pool-server_config-bc1b
2021-02-09 14:35:15,802 test             L0779 INFO | PASS 4-./object/obj_open_bad_param.py:ObjOpenBadParam.test_pool_handle_as_obj_handle;hosts-pool-server_config-bc1b
2021-02-09 14:35:28,612 test             L0779 INFO | PASS 5-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_ranklist;hosts-pool-server_config-bc1b
2021-02-09 14:35:41,413 test             L0779 INFO | PASS 6-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_oid;hosts-pool-server_config-bc1b
2021-02-09 14:35:54,746 test             L0779 INFO | PASS 7-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_tgts;hosts-pool-server_config-bc1b
2021-02-09 14:36:07,706 test             L0779 INFO | PASS 8-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_attrs;hosts-pool-server_config-bc1b

2021-02-09 14:34:59,260 test             L0139 INFO | *** SETUP running on <LinuxDistro: name=SuSE, version=15, release=2, arch=x86_64> ***
2021-02-09 14:35:36,560 test             L0779 INFO | PASS 1-./object/obj_open_bad_param.py:ObjOpenBadParam.test_bad_obj_handle;hosts-pool-server_config-c281
2021-02-09 14:35:56,799 test             L0779 INFO | PASS 2-./object/obj_open_bad_param.py:ObjOpenBadParam.test_invalid_container_handle;hosts-pool-server_config-c281
2021-02-09 14:36:11,983 test             L0779 INFO | PASS 3-./object/obj_open_bad_param.py:ObjOpenBadParam.test_closed_container_handle;hosts-pool-server_config-c281
2021-02-09 14:36:26,924 test             L0779 INFO | PASS 4-./object/obj_open_bad_param.py:ObjOpenBadParam.test_pool_handle_as_obj_handle;hosts-pool-server_config-c281
2021-02-09 14:36:51,651 test             L0779 INFO | PASS 5-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_ranklist;hosts-pool-server_config-c281
2021-02-09 14:37:06,491 test             L0779 INFO | PASS 6-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_oid;hosts-pool-server_config-c281
2021-02-09 14:37:21,343 test             L0779 INFO | PASS 7-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_tgts;hosts-pool-server_config-c281
2021-02-09 14:37:36,420 test             L0779 INFO | PASS 8-./object/obj_open_bad_param.py:ObjOpenBadParam.test_null_attrs;hosts-pool-server_config-c281

**Comment by cibervicho** [comment]
The build errors shown as failing in this PR were a transient network issue. I re-ran the Build phase from BO and all passed: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-4586/6/pipeline

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -306,7 +306,10 @@ obj_layout_create(struct dc_object *obj, bool refresh)
 	int			rc;
 
 	pool = dc_hdl2pool(dc_cont_hdl2pool_hdl(obj->cob_coh));
-	D_ASSERT(pool != NULL);
+	if (pool == NULL) {
+		D_DEBUG(DB_PL, "Cannot find valid pool\n");
```
Comment (berserk-fury) [inline src/object/cli_obj.c:310]: Perhaps D_WARN ?

**Review by jolivier23 — APPROVED** [review]

**Review by berserk-fury — APPROVED** [review]

**Comment by cibervicho** [comment]
Tests passed CI checks:

2021-02-11 03:23:52,450 test             L0139 INFO | *** SETUP running on <LinuxDistro: name=centos, version=7, release=9, arch=x86_64> ***
2021-02-11 03:23:52,238 test             L0779 INFO | PASS 1-./object/obj_open_bad_param.py:ObjOpenBadParam.test_invalid_container_handle;hosts-pool-server_config-07d4
2021-02-11 03:24:05,062 test             L0779 INFO | PASS 2-./object/obj_open_bad_param.py:ObjOpenBadParam.test_closed_container_handle;hosts-pool-server_config-07d4

2021-02-11 03:24:29,228 test             L0139 INFO | *** SETUP running on <LinuxDistro: name=SuSE, version=15, release=2, arch=x86_64> ***
2021-02-11 03:25:06,548 test             L0779 INFO | PASS 1-./object/obj_open_bad_param.py:ObjOpenBadParam.test_invalid_container_handle;hosts-pool-server_config-0d9b
2021-02-11 03:25:21,138 test             L0779 INFO | PASS 2-./object/obj_open_bad_param.py:ObjOpenBadParam.test_closed_container_handle;hosts-pool-server_config-0d9b

**Review by sylviachanoiyee — CHANGES_REQUESTED** [review]
David, this change is in a common file used by a lot of tests, Could you rerun all test stages with no pragma filter? Thanks!

**Comment by cibervicho** [comment]
> David, this change is in a common file used by a lot of tests, Could you rerun all test stages with no pragma filter? Thanks!

Sure, doing it right now. Thank you.

**Comment by cibervicho** [comment]
CI checks just finished without errors reported

**Pull request merged**